### DISCUSSION TITLE
Advanced capture: Dockerfile with ARG

### DISCUSSION THREAD

**Discussion by andormarkus** [renovatebot/renovate#10953] [Request Help]
Hi All,

Based on the [documentation](advanced-capture) Renovate can update the `ENV` of the Dockerfile if advanced capture is enabled. However we are using `ARG` for environment variable declaration becase we need them only for dockerfile build.

Does Renovate support `ARG` as well or only `ENV`?

Thanks,
Andor

![image](https://user-images.githubusercontent.com/51825189/126865096-212fd301-69cd-4cde-bcb3-9843067d1db9.png)

**Comment by rarkins** [comment]
Please provide a minimum example Dockerfile so we can look

**Reply by andormarkus** [reply]
```Dockerfile
FROM python:3.8-slim-buster

# renovate: datasource=github-releases depName=aquasecurity/tfsec
ARG TFSEC_VERSION="v0.51.2"

# Install general dependencies
RUN apt update && \
    apt install -y curl gnupg gawk unzip software-properties-common

RUN curl -L "$(curl -s https://api.github.com/repos/aquasecurity/tfsec/releases | grep -o -E -m 1 "https://.+?/${TFSEC_VERSION}/tfsec-linux-amd64")" > tfsec && chmod +x tfsec && mv tfsec /usr/bin/

RUN tfsec --help
```

**Reply by andormarkus** [reply]
Based on your answer, it would be easier to rename all the `ARG` to `ENV` and we can avoid  Renovate's "regex" manager.
Am I correct?

**Reply by rarkins** [reply]
Sorry, my previous answer was wong because somehow I missed the `# renovate:` line when reading from mobile. I have deleted it so it's not misleading.

It's possible that you can reuse this preset we have ourselves: https://github.com/renovatebot/.github/blob/4a1aab11ab13676adc5440cf41cd1f9dd9479c50/default.json#L108-L114

Although you might need to remove the `"` quotation marks from the value or alter our regex to add them outside the `currentValue` capture.
